scipy/stats/_morestats.py

```diff
@@ -3027,19 +3027,29 @@ def mood(x, y, axis=0, alternative="two-sided"):
 
 
 def wilcoxon_result_unpacker(res):
+    if res.zstatistic is None:
+        return res.statistic, res.pvalue
     return res.statistic, res.pvalue, res.zstatistic
 
 
-def wilcoxon_result_object(statistic, pvalue, zstatistic):
+def wilcoxon_result_object(statistic, pvalue, zstatistic=None):
     return WilcoxonResult(statistic, pvalue, zstatistic=zstatistic)
 
 
+def wilcoxon_outputs(kwds):
+    method = kwds.get('method', 'auto')
+    x = kwds.get('x')
+    if method == 'exact' or (method == 'auto' and len(x) <= 50):
```

Comment:
IIRC `x` is an ND array at this point rather than a slice. so we would need to check the length along `axis`.